hyperspy/models/model1d.py

```diff
@@ -710,15 +710,36 @@ def _model2plot(self, axes_manager, out_of_range2nans=True):
             ns[np.where(self.channel_switches)] = s
             s = ns
         return s
-
-    def plot(self, plot_components=False, **kwargs):
+    
+    def _residual2plot(self, axes_manager, out_of_range2nans=True):
+        old_axes_manager = None
+        if axes_manager is not self.axes_manager:
+            old_axes_manager = self.axes_manager
+            self.axes_manager = axes_manager
+            self.fetch_stored_values()
+        #Residual = Signal - model
+        s = (self.signal.__call__() -
+            self.__call__(non_convolved=False, onlyactive=True))
+        if old_axes_manager is not None:
+            self.axes_manager = old_axes_manager
+            self.fetch_stored_values()
+        if out_of_range2nans is True:
```

Comment:
Since there currently is no way to change the value of `out_of_range2nans` from a calling function, I suggest you just remove the `if` and always use the nan-behaviour.

If you do keep the `if`, then you can remove `is True`, since `if out_of_range2nans` will evaluate to `if True` or `if False`.